internal/js/modules/k6/browser/browser/mouse_mapping.go

```diff
@@ -9,35 +11,50 @@ import (
 
 func mapMouse(vu moduleVU, m *common.Mouse) mapping {
 	return mapping{
-		"click": func(x float64, y float64, opts sobek.Value) *sobek.Promise {
+		"click": func(x float64, y float64, opts sobek.Value) (*sobek.Promise, error) {
+			popts := common.NewMouseClickOptions()
+			if err := popts.Parse(vu.Context(), opts); err != nil {
+				return nil, fmt.Errorf("parsing mouse click options: %w", err)
+			}
 			return k6ext.Promise(vu.Context(), func() (any, error) {
-				// TODO(@mstoykov): don't use sobek Values in a separate goroutine
-				return nil, m.Click(x, y, opts) //nolint:wrapcheck
-			})
+				return nil, m.Click(x, y, popts) //nolint:wrapcheck
+			}), nil
 		},
-		"dblClick": func(x float64, y float64, opts sobek.Value) *sobek.Promise {
+		"dblClick": func(x float64, y float64, opts sobek.Value) (*sobek.Promise, error) {
+			popts := common.NewMouseDblClickOptions()
+			if err := popts.Parse(vu.Context(), opts); err != nil {
+				return nil, fmt.Errorf("parsing double click options: %w", err)
+			}
 			return k6ext.Promise(vu.Context(), func() (any, error) {
-				// TODO(@mstoykov): don't use sobek Values in a separate goroutine
-				return nil, m.DblClick(x, y, opts) //nolint:wrapcheck
-			})
+				return nil, m.DblClick(x, y, popts) //nolint:wrapcheck
+			}), nil
 		},
-		"down": func(opts sobek.Value) *sobek.Promise {
+		"down": func(opts sobek.Value) (*sobek.Promise, error) {
+			popts := common.NewMouseDownUpOptions()
+			if err := popts.Parse(vu.Context(), opts); err != nil {
+				return nil, fmt.Errorf("parsing mouse down options: %w", err)
+			}
 			return k6ext.Promise(vu.Context(), func() (any, error) {
-				// TODO(@mstoykov): don't use sobek Values in a separate goroutine
-				return nil, m.Down(opts) //nolint:wrapcheck
-			})
+				return nil, m.Down(popts) //nolint:wrapcheck
+			}), nil
 		},
-		"up": func(opts sobek.Value) *sobek.Promise {
+		"up": func(opts sobek.Value) (*sobek.Promise, error) {
+			popts := common.NewMouseDownUpOptions()
+			if err := popts.Parse(vu.Context(), opts); err != nil {
+				return nil, fmt.Errorf("parsing mouse down options: %w", err)
```

Comment:
```suggestion
				return nil, fmt.Errorf("parsing mouse up options: %w", err)
```

Or maybe even call it up/down :shrug: